src/routes/achievementRoutes.js

```diff
@@ -26,14 +27,18 @@ router.post(
 router.get(
   '/',
   achievementController.getAllAchievements,
+);
+
+router.get(
+  '/:id',
   achievementController.getAchievementById,
 );
 
-router.put(
-  '/',
+router.patch(
+  '/:id',
   authentication(),
   checkUpdateFile('logo'),
-  postAchievementValidationRules(),
+  updateAchievementValidationRules(),
   validate,
   achievementController.updateAchievementById,
 );
```

Comment:
Please make the `router.put` still intact and create a new `router.patch` without deleting the `router.put`